pkg/report/table/summary.go

```diff
@@ -0,0 +1,308 @@
+		tml.DisableFormatting()
+	}
+
+	var ss []Scanner
+	for _, scanner := range scanners {
+		s := NewScanner(scanner)
+		if lo.IsNil(s) {
+			continue
+		}
+		ss = append(ss, s)
+	}
+
+	return &summaryRenderer{
+		w:          buf,
+		isTerminal: isTerminal,
+		scanners:   ss,
+	}
+}
+
+func (r *summaryRenderer) Render(report types.Report) {
+	if len(r.scanners) == 0 {
+		log.WithPrefix("report").Warn("No enabled scanners found. Summary table will not be displayed.")
+		return
+	}
+
+	r.printf("\n<underline><bold>Report Summary</bold></underline>\n\n")
+
+	t := newTableWriter(r.w, r.isTerminal)
+	t.SetAutoMerge(false)
+	t.SetColumnMaxWidth(80)
+
+	headers := []string{
+		"Target",
+		"Type",
+	}
+	alignments := []table.Alignment{
+		table.AlignLeft,
+		table.AlignCenter,
+	}
+	for _, scanner := range r.scanners {
+		headers = append(headers, scanner.Header())
+		alignments = append(alignments, scanner.Alignment())
+	}
+	t.SetHeaders(headers...)
+	t.SetAlignment(alignments...)
+
+	for _, result := range splitAggregatedPackages(report.Results) {
+		resultType := string(result.Type)
+		if result.Class == types.ClassSecret {
+			resultType = "text"
+		} else if result.Class == types.ClassLicense || result.Class == types.ClassLicenseFile {
+			resultType = "-"
+		}
+		rows := []string{
+			result.Target,
+			resultType,
+		}
+		for _, scanner := range r.scanners {
+			rows = append(rows, r.colorizeCount(scanner.Count(result)))
+		}
+		t.AddRows(rows)
+	}
+
+	if len(report.Results) == 0 {
+		r.showEmptyResultsWarning()
+		t.AddRows(slices.Repeat([]string{"-"}, len(r.scanners)+2))
+	}
+
+	t.Render()
+
+	// Show legend
+	r.printf("Legend:\n" +
+		"- '-': Not scanned\n" +
+		"- '0': Clean (no security findings detected)\n\n")
+
+	return
+}
+
+func (r *summaryRenderer) printf(format string, args ...any) {
+	// nolint
+	_ = tml.Fprintf(r.w, format, args...)
+}
+
+// showEmptyResultsWarning shows WARN why the results array is empty based on the enabled scanners.
+// We need to separate the vuln/misconfig and secret/license scanners,
+// because the results array contains results without findings for vulns/misconfig only.
+func (r *summaryRenderer) showEmptyResultsWarning() {
+	resultByFiles := []Scanner{
+		NewScanner(types.VulnerabilityScanner),
+		NewScanner(types.MisconfigScanner),
+	}
+	resultByFindings := []Scanner{
+		NewScanner(types.SecretScanner),
+		NewScanner(types.LicenseScanner),
+	}
+
+	var warnStrings []string
+	if scanners := lo.Intersect(resultByFiles, r.scanners); len(scanners) > 0 {
+		warnStrings = append(warnStrings, fmt.Sprintf("Supported files for %s scanner(s) not found.",
+			strings.Join(xstrings.ToStringSlice(scanners), "/")))
+	}
+	if scanners := lo.Intersect(resultByFindings, r.scanners); len(scanners) > 0 {
+		warnStrings = append(warnStrings, fmt.Sprintf("No results found for %s scanner(s).",
+			strings.Join(xstrings.ToStringSlice(scanners), "/")))
+	}
+
+	log.WithPrefix("report").Info(strings.Join(warnStrings, " "))
```

Comment:
```suggestion
	if scanners := lo.Intersect(resultByFiles, r.scanners); len(scanners) > 0 {
		r.logger.Warn("Supported files for scanner(s) not found.", log.Any("scanners", scanners))
	}
	if scanners := lo.Intersect(resultByFindings, r.scanners); len(scanners) > 0 {
		r.logger.Warn("No results found for scanner(s).", log.Any("scanners", scanners))
	}
```

Reply:
changed in https://github.com/aquasecurity/trivy/pull/8177/commits/67a729cd2b36e3171316c26133da2e5a15c8b9a2